openstates/al/bills.py

```diff
@@ -39,222 +43,357 @@ def _categorize_action(action):
+        (viewstategenerator, ) = doc.xpath('//input[@id="__VIEWSTATEGENERATOR"]/@value')
+        form = {
+                '__EVENTTARGET': 'ctl00$cboSession',
+                '__EVENTARGUMENT': '',
+                '__LASTFOCUS': '',
+                '__VIEWSTATE': viewstate,
+                '__VIEWSTATEGENERATOR': viewstategenerator,
+                'ctl00$cboSession': self.session_name
+                }
+        self.spost(url=SESSION_SET_URL, data=form, allow_redirects=False)
+
+    def scrape(self, session, chambers):
+        self.session = session
+        self.session_name = self.metadata['session_details'][self.session]['_scraped_name']
+        self.session_id = self.metadata['session_details'][self.session]['internal_id']
+
+        self._set_session(session)
+
+        # Acquire and process a list of all bills
+        BILL_TYPE_URL = 'http://alisondb.legislature.state.al.us/Alison/SESSBillsBySelectedStatus.aspx'
+        BILL_LIST_URL = 'http://alisondb.legislature.state.al.us/Alison/SESSBillsList.aspx?STATUSCODES=Had%20First%20Reading%20House%20of%20Origin&BODY=999999'
+
+        doc = lxml.html.fromstring(self.sget(url=BILL_TYPE_URL).text)
+        (viewstate, ) = doc.xpath('//input[@id="__VIEWSTATE"]/@value')
+        (viewstategenerator, ) = doc.xpath('//input[@id="__VIEWSTATEGENERATOR"]/@value')
+        form = {
+                '__EVENTTARGET': 'ctl00$MainDefaultContent$gvStatus$ctl02$ctl00',
+                '__EVENTARGUMENT': 'Select$0',
+                '__LASTFOCUS': '',
+                '__VIEWSTATE': viewstate,
+                '__VIEWSTATEGENERATOR': viewstategenerator,
+                'ctl00$cboSession': self.session_name,
+                'ctl00$ScriptManager1': 'ctl00$UpdatePanel1|ctl00$MainDefaultContent$gvStatus$ctl02$ctl00'
+        }
+        self.spost(url=BILL_TYPE_URL, data=form, allow_redirects=False)
+
+        self.scrape_bill_list(BILL_LIST_URL)
+
+        # Acquire and process a list of all resolutions
+        RESOLUTION_TYPE_URL = 'http://alisondb.legislature.state.al.us/Alison/SESSResBySelectedStatus.aspx'
+        RESOLUTION_LIST_URL = 'http://alisondb.legislature.state.al.us/Alison/SESSResList.aspx?STATUSCODES=Had%20First%20Reading%20House%20of%20Origin&BODY=999999'
+
+        doc = lxml.html.fromstring(self.sget(url=RESOLUTION_TYPE_URL).text)
+        (viewstate, ) = doc.xpath('//input[@id="__VIEWSTATE"]/@value')
+        (viewstategenerator, ) = doc.xpath('//input[@id="__VIEWSTATEGENERATOR"]/@value')
+        form = {
+                '__EVENTTARGET': 'ctl00$MainDefaultContent$gvStatus$ctl02$ctl00',
+                '__EVENTARGUMENT': 'Select$0',
+                '__LASTFOCUS': '',
+                '__VIEWSTATE': viewstate,
+                '__VIEWSTATEGENERATOR': viewstategenerator,
+                'ctl00$cboSession': self.session_name,
+                'ctl00$ScriptManager1': 'ctl00$UpdatePanel1|ctl00$MainDefaultContent$gvStatus$ctl02$ctl00'
+        }
```

Comment:
I definitely see where you're coming from, but I feel that since it only comes up a few times and the form is significantly different each time (with some common keys and values but most not following a pattern), it wouldn't make the code cleaner to modularize it into a function.

That said, I'm open for sure to executing this idea if you think it'd be a net benefit.